x/stakeibc/keeper/unbonding_records.go

```diff
@@ -96,6 +111,74 @@ func (k Keeper) GetTotalUnbondAmountAndRecordsIds(ctx sdk.Context, chainId strin
 	return totalUnbonded, unbondingRecordIds
 }
 
+// Update Native Amounts in the Host Zone Unbonding Record (and associated User Redemption Records) to reflect the current redemption rate
+// If setNativeAmountsToZero is true, then the native amounts are set to zero, ignoring the redemption rate. This should be used if the unbonding didn't succeed
+// Otherwise, the native amounts are set based on the current redemption rate
+func (k Keeper) UpdateNativeTokensForHostZoneUnbondingRecord(
+	ctx sdk.Context,
+	epochNumber uint64,
+	hostZoneUnbondingRecord *recordstypes.HostZoneUnbonding,
+	setNativeAmountsToZero bool,
+) bool {
+	totalNativeAmount := sdkmath.ZeroInt()
+	chainId := hostZoneUnbondingRecord.GetHostZoneId()
+	hostZone, found := k.GetHostZone(ctx, chainId)
+	if !found {
+		k.Logger(ctx).Info(utils.LogWithHostZone(chainId, "No host zone found for id %s, Not Updating Native Token Amounts", chainId))
+		return false
+	}
+	redemptionRate := hostZone.RedemptionRate
+	// loop through user redemption records for the host zone unbonding record, update each one
+	for _, userRedemptionRecordId := range hostZoneUnbondingRecord.GetUserRedemptionRecords() {
+		userRedemptionRecord, found := k.RecordsKeeper.GetUserRedemptionRecord(ctx, userRedemptionRecordId)
+		if !found {
+			k.Logger(ctx).Info(utils.LogWithHostZone(chainId, "No user redemption record found for id %s", userRedemptionRecordId))
+			continue
+		}
+		// update the amount, based on the current redemption rate (or set to zero if desired)
+		nativeAmount := sdkmath.ZeroInt()
+		if !setNativeAmountsToZero {
```

Comment:
imo we shouldn't mix this into the function - we should have 
```
UpdateNativeTokensForHostZoneUnbondingRecord
ResetNativeTokensForHostZoneUnbondingRecord
```
and these could optionally call a sub function

the current setup of `SetNativeTokensToZeroInUnbondingRecords` calling `UpdateNativeTokensForHostZoneUnbondingRecord` is confusing